### PR TITLE
chore: upgrade to Laravel 9

### PR DESCRIPTION
**Pull request by crnkovic**
## Summary

https://app.clickup.com/t/2nh84uy

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: tests/Unit/Services/ExchangeRateTest.php [hunk 1]
```diff
@@ -22,10 +22,10 @@
...
 it('should convert with the current rate', function () {
     (new CryptoDataCache())->setPrices('USD.day', collect([
-        Carbon::now('-3 hours')->format('Y-m-d H:i:s') => 1,
-        Carbon::now('-2 hours')->format('Y-m-d H:i:s') => 2,
-        Carbon::now('-1 hour')->format('Y-m-d H:i:s')  => 3,
-        Carbon::now()->format('Y-m-d H:i:s')           => 10,
+        Carbon::parse('-3 hours')->format('Y-m-d H:i:s') => 1,
+        Carbon::parse('-2 hours')->format('Y-m-d H:i:s') => 2,
+        Carbon::parse('-1 hour')->format('Y-m-d H:i:s')  => 3,
+        Carbon::now()->format('Y-m-d H:i:s')             => 10,
```
Comment (crnkovic) [inline tests/Unit/Services/ExchangeRateTest.php:28]: `Carbon#now` now accepts timezone as a first argument. `Carbon#parse` can be used to indicate we want something 3 hours ago.

File: app/Console/Commands/CacheDelegateResignationIds.php [hunk 2]
```diff
@@ -31,6 +31,9 @@ public function handle(): void
             ->select('sender_public_key', 'id')
             ->withScope(DelegateResignationScope::class)
             ->cursor()
-            ->each(fn ($transaction) => CacheResignationId::dispatch($transaction->sender_public_key, (string) $transaction->id)->onQueue('resignations'));
+            ->each(function ($transaction) {
+                // @phpstan-ignore-next-line
```
Comment (crnkovic) [inline app/Console/Commands/CacheDelegateResignationIds.php:35]: Ignored because PHPStan reports an error that `$transaction->id` is already a string... but it's not, it's an integer and since we have strict types removing `(string)` would report an exception.

File: app/Console/Commands/CacheFees.php [hunk 3]
```diff
@@ -43,12 +43,12 @@ public function handle(FeeCache $cache): void
             $cache->setHistorical($period, HistoricalAggregateFactory::make($period)->aggregate());
         });
 
-        collect(Forms::getTransactionOptions())->except('all')->keys()->each(function ($type) use ($cache): void {
+        collect(Forms::getTransactionOptions())->except(['all'])->keys()->each(function ($type) use ($cache): void {
             preg_match('/^[a-z]+(\d+)$/', self::LAST_20, $match);
 
             $result = (new LastFeeAggregate())
                 ->setLimit((int) $match[1])
-                ->setType($type ?? '')
```
Comment (crnkovic) [inline app/Console/Commands/CacheFees.php:51]: `$type` is already non-nullable string, so null-coalescing operator `??` has no effect.

File: app/Services/Transactions/Aggregates/Fees/LastFeeAggregate.php [hunk 4]
```diff
@@ -42,9 +42,9 @@ public function aggregate(): array
             ->map(fn ($fee) => $fee->toFloat());
 
         return [
-            'minimum' => $fees->min() ?? 0,
+            'minimum' => $fees->min() ?? 0, // @phpstan-ignore-line
```
Comment (crnkovic) [inline app/Services/Transactions/Aggregates/Fees/LastFeeAggregate.php:45]: Ignored because `min()` and `max()` supposedly never return 0 (or as is thought by PHPStan), however tests fail when null-coalescing operator is removed.

File: app/Console/Commands/CacheDelegateResignationIds.php (hunk 2 above)
Comment (ItsANameToo) [reply app/Console/Commands/CacheDelegateResignationIds.php:35]: isn't it assuming it's a string because of the `->select('sender_public_key', 'id')`? could we add an explicit type to the collection it returns so phpstan understands what `$transaction` is exactly instead of defaulting to `mixed` for it? I didn't get the string error when I removed the phpstan line, instead it doesn't seem to understand that an object is returned with two properties:

```
  Line   Console/Commands/CacheDelegateResignationIds.php
 ------ ------------------------------------------------------------
  35     Cannot access property $id on (int|string).
  35     Cannot access property $sender_public_key on (int|string).
  ```

File: app/Console/Commands/CacheFees.php (hunk 3 above)
Comment (ItsANameToo) [reply app/Console/Commands/CacheFees.php:51]: could also change it to `->each(function (string $type)` to make it explicit that it's not null 🤔

File: app/Console/Commands/CacheDelegateResignationIds.php (hunk 2 above)
Comment (crnkovic) [reply app/Console/Commands/CacheDelegateResignationIds.php:35]: A bit poor choice of words on my part, let me explain in more detail:

If we add typehint that `$transaction` is a Transaction object, it will complain about "Casting something to string that's already a string". It will know that `$transaction` is of Transaction object, but Transaction model at the top has a model property declaration that `$id` is a string, and a `TransactionViewModel` has `id(): string` method. I could change that typehint and return type to `int`, but then the tests will fail:

```
App\ViewModels\TransactionViewModel::id(): Return value must be of type int, string returned
```

What I mean:

- [TransactionViewModel#id returning a string. Setting that to `int` fails some tests](https://github.com/ArdentHQ/arkscan/blob/develop/app/ViewModels/TransactionViewModel.php#L63)
- [Transaction model indicating that ID is `string`. Changing that to `int` fails static analysis because of `TransactionViewModel`](https://github.com/ArdentHQ/arkscan/blob/develop/app/Models/Transaction.php#L29)

If we remove `(string)` cast, static analysis will output:

```
36     Parameter #2 $transactionId of job class App\Jobs\CacheResignationId constructor expects string in App\Jobs\CacheResignationId::dispatch(), int given.
```

Because that job class expects an int.

Basically, TLDR is:

- If I switch everything down the line to `int` -- tests will fail that "string is expected", static analysis passes
- If I switch everything down the line to `string` -- some logical test will fail (` Failed asserting that 50 matches expected 153.`) and static analysis fails

I can add TODO to that phpstan-ignore to note that I should take a look at that because I don't want to go over why test fail when changing typehints in this PR that's dedicated to Laravel 9.

**Review by ItsANameToo — APPROVED** [review]
